cmd/containerd-shim-runhcs-v1/pod.go

```diff
@@ -126,6 +126,10 @@ func createPod(ctx context.Context, events publisher, req *task.CreateTaskReques
 			parent.Close()
 			return nil, err
 		}
+		if err := oci.HandleCPUGroupSetup(ctx, parent, s.Annotations); err != nil {
```

Comment:
I would do this as part of `uvm.CreateWCOW` and `uvm.CreateLCOW`. Passing the group ID in via the UVM options seems like a good fit.

Reply:
It also lets us parse the annotation in the same place as we do the other UVM annotations.

Reply:
The issue is that we can't set the cpugroup until after the VM has started, so we wouldn't be able to do anything in `uvm.Create*COW`

Reply:
If we could find a way to assign the vm to the group in `Start` itself that would be nice. Only (ugly) way I could think of is in Create*COW assign UtilityVM.cpuGroupID to the group ID passed in via the create opts and then check if this isn't 0 in `Start` and do the setup, which is obviously gross. That variable being set should mean it's already successfully been assigned to the group not as a temp until it hopefully is.

Reply:
I think this would be good. I think needing to set the CPU group on start is basically an implementation detail, so I would suggest we:
- Pass in the CPU group ID in the UVM options
- Save the ID in the UVM struct
- At start time, set the CPU group if an ID is set on the struct

Reply:
I like that plan, will update

Reply:
Done